### DISCUSSION TITLE
How can I search by included models in one "where"?

### DISCUSSION THREAD

**Discussion by Evernayt** [sequelize/sequelize#15725] [Help & Questions] [answered]
I use the given "where" to search:
```js
where = {
  [Op.or]: [
    {
      name: {
        [Op.like]: `%${word}%`,
      },
    },
    {
      description: {
        [Op.like]: `%${word}%`,
      },
    },
    {
      '$types.name$': {
        [Op.like]: `%${word}%`,
      },
    },
  ],
};
```
And a method for getting paginated `products`, which should include the `Type` and `Feature` models:
```js
const products = await this.productModel.findAndCountAll({
  subQuery: false,
  limit,
  offset,
  distinct: true,
  order: [['name', 'ASC']],
  where,
  include: [
    {
      model: Type,
      include: [
        {
          model: Feature,
        },
      ],
    },
  ],
});
```
I have to use `subQuery: false` for `'$types.name$'` to work. But `subQuery: false` doesn't work correctly with `limit` and `offset` (I don't get all the data).

Are there ways to fix this?

Product model:
```js
@Table({ tableName: 'products' })
export class Product extends Model<Product, ProductCreationAttrs> {
  @Column({
    type: DataType.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  })
  id: number;

  @Column({ type: DataType.STRING, unique: true, allowNull: false })
  name: string;

  @Column({ type: DataType.STRING, allowNull: false })
  description: string;

  @HasMany(() => Type, { foreignKey: 'productId' })
  types: Type[];
}
```
Type model:
```js
@Table({ tableName: 'types' })
export class Type extends Model<Type, TypeCreationAttrs> {
  @Column({
    type: DataType.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  })
  id: number;

  @Column({ type: DataType.STRING, allowNull: false })
  name: string;

  @BelongsTo(() => Product, { foreignKey: 'productId' })
  product: Product;

  @BelongsToMany(() => Feature, () => TypeFeatures)
  features: Feature[];
}
```
Dialect: MariaDB

**Comment by ephys** [comment, marked as answer]
If you use `limit`, Sequelize will produce a subquery to apply the limit only to your top level model (in this instance, Product)

This means that you can't reference child association attributes anymore

`subquery:false` forces Sequelize to revert back to a JOIN, which makes it possible to reference association attributes again, but makes the limit apply to the entire query

There is no way around that. The [future rewrite](https://github.com/sequelize/sequelize/issues/15260) will make these limitations clearer but won't be able to remove them, as we're limited by SQL itself.

You need to use an alternative solution to reach your goals. Maybe using a subquery in your where condition may fit your needs. Alternatively try writing a query in raw sql using `sequelize.query`

**Reply by nicoabie** [reply]
this post explains how to get over that limitation using a select distinct of the ids

https://github.com/sequelize/sequelize/issues/15650
